hls4ml/templates/oneapi/firmware/nnet_utils/nnet_depthconv2d_resource.h

```diff
@@ -0,0 +1,61 @@
+#ifndef NNET_SEPARABLE_CONV2D_LATENCY_H_
+#define NNET_SEPARABLE_CONV2D_LATENCY_H_
+
+#include "nnet_common.h"
+#include "nnet_conv2d_resource.h"
+#include "nnet_mult.h"
+
+namespace nnet {
+
+template <class data_T, class res_T, typename CONFIG_T>
+void depthwise_conv_2d_resource_cl(const data_T &data, res_T &res, const typename CONFIG_T::weight_t &weights,
+                                   const typename CONFIG_T::bias_t &biases) {
+
+    int depth_multiplier = CONFIG_T::n_filt / CONFIG_T::n_chan;
+    [[intel::fpga_register]] int res_idx = 0;
+
+DM_LOOP:
```

Comment:
The Xilinx version, https://github.com/fastmachinelearning/hls4ml/blob/main/hls4ml/templates/vivado/nnet_utils/nnet_sepconv2d_latency.h, seems to have fewer nested loops, even if you take into account the extra depth-multiplier loop here. Do we understand the difference?

Reply:
This oneAPI implementation is very close to the catapult implementation. I'm not familiar with the Xilinx one.